app/scripts/services/metrics.js

```diff
@@ -2,15 +2,27 @@
 
 angular.module("openshiftConsole")
   .factory("MetricsService", function($http, $q, APIDiscovery) {
-    var COUNTER_TEMPLATE = "/counters/{containerName}%2F{podUID}%2F{metric}/data";
-    var GAUGE_TEMPLATE = "/gauges/{containerName}%2F{podUID}%2F{metric}/data";
+    var POD_COUNTER_TEMPLATE = "/counters/{containerName}%2F{podUID}%2F{metric}/data";
+    var POD_GAUGE_TEMPLATE = "/gauges/{containerName}%2F{podUID}%2F{metric}/data";
+
+    // Use a regex to match the label deployement=<name> at word boundaries. In Hawkular, it's
+    // stored as a comma-separated list of values in the form name:value.
+    var DEPLOYMENT_COUNTER_TEMPLATE = "/counters/data?stacked=true&tags=descriptor_name:{metric},type:{type},labels:.*\\bdeployment:{deployment}\\b.*";
+    var DEPLOYMENT_GAUGE_TEMPLATE = "/gauges/data?stacked=true&tags=descriptor_name:{metric},type:{type},labels:.*\\bdeployment:{deployment}\\b.*";
 
     // URL template to show for each type of metric.
     var templateByMetric = {
```

Comment:
deploymentURLTemplateByMetric and podURLTemplateByMetric ?  longer but you
avoid the confusion

On Thu, May 19, 2016 at 1:46 PM, Sam Padgett <notifications@github.com>
wrote:

> In app/scripts/services/metrics.js
> <https://github.com/openshift/origin-web-console/pull/22#discussion_r63923489>
> :
>
> > @@ -2,15 +2,27 @@
> >
> >  angular.module("openshiftConsole")
> >    .factory("MetricsService", function($http, $q, APIDiscovery) {
> > -    var COUNTER_TEMPLATE = "/counters/{containerName}%2F{podUID}%2F{metric}/data";
> > -    var GAUGE_TEMPLATE = "/gauges/{containerName}%2F{podUID}%2F{metric}/data";
> > +    var POD_COUNTER_TEMPLATE = "/counters/{containerName}%2F{podUID}%2F{metric}/data";
> > +    var POD_GAUGE_TEMPLATE = "/gauges/{containerName}%2F{podUID}%2F{metric}/data";
> > +
> > +    // Use a regex to match the label deployement=<name> at word boundaries. In Hawkular, it's
> > +    // stored as a comma-separated list of values in the form name:value.
> > +    var DEPLOYMENT_COUNTER_TEMPLATE = "/counters/data?stacked=true&tags=descriptor_name:{metric},type:{type},labels:.*\\bdeployment:{deployment}\\b.*";
> > +    var DEPLOYMENT_GAUGE_TEMPLATE = "/gauges/data?stacked=true&tags=descriptor_name:{metric},type:{type},labels:.*\\bdeployment:{deployment}\\b.*";
> >
> >      // URL template to show for each type of metric.
> >      var templateByMetric = {
>
> I wasn't sure if I should rename the variable to avoid confusion with "pod
> template." I'll change it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/22/files/4e3d752806598597cae3557832988b7c94e98c31#r63923489>
>